icicle/backend/cpu/src/field/cpu_vec_ops.cpp

```diff
@@ -803,12 +803,18 @@ REGISTER_SLICE_EXT_FIELD_BACKEND("CPU", cpu_slice<extension_t>);
 
 /*********************************** Highest non-zero idx ***********************************/
 template <typename T>
-eIcicleError cpu_highest_non_zero_idx(
-  const Device& device, const T* input, uint64_t size, const VecOpsConfig& config, int64_t* out_idx /*OUT*/)
+eIcicleError cpu_highest_non_zero_index(
+  const Device& device,
+  const T* input,
+  uint64_t size,
+  const VecOpsConfig& config,
+  int64_t* out_idx /*OUT*/,
+  int32_t idx_in_batch_to_calc)
 {
   ICICLE_ASSERT(input && out_idx && size != 0) << "Error: Invalid argument";
   uint64_t stride = config.columns_batch ? config.batch_size : 1;
   for (uint64_t idx_in_batch = 0; idx_in_batch < config.batch_size; ++idx_in_batch) {
+    if (idx_in_batch_to_calc != -1 && idx_in_batch_to_calc != idx_in_batch) { continue; }
```

Comment:
better compute the start->end indices of the loop to be either a specific index or 0 to batch_size instead of doing many iterations and skipping all except one.